### PR TITLE
Add Django Prometheus

### PR DESCRIPTION
**Pull request by jb3**
Adds Django Prometheus and installs the necessary middleware and URL paths for reporting.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -21,6 +21,7 @@ gunicorn = "~=20.0.4"
 sentry-sdk = "~=0.19"
 markdown = "~=3.3.4"
 python-frontmatter = "~=1.0"
+django-prometheus = "^2.1.0"
```
Comment (doublevcodes) [inline pyproject.toml]:
```suggestion
django-prometheus = "~=2.1"
```
This should be changed to follow the format of the other pinned dependencies.

**Review by doublevcodes — CHANGES_REQUESTED** [review]

**Review by doublevcodes — APPROVED** [review]
LGTM now :shipit:

**Review by Den4200 — APPROVED** [review]

**Review by jchristgit — CHANGES_REQUESTED** [review]
Yes, but also no.

This is nice. This needs a change though, because we are using multiple workers, because Python can't gobble around on more than one core in the same address space.

`gunicorn` needs to be started with a configuration file as its argument, with the following contents:
```py
from prometheus_client import multiprocess

def child_exit(server, worker):
    multiprocess.mark_process_dead(worker.pid)
```

This is step three from the Prometheus Python client's multiprocess mode considerations: https://github.com/prometheus/client_python. Step two is [taken care of](https://github.com/korfuri/django-prometheus/blob/434a3ba36bdada45c9633451f5f6cfd145814ccf/django_prometheus/exports.py#L119) by `django-prometheus`, step one I will PR into the Kubernetes repository.

**Comment by jchristgit** [comment]
And as a small bonus, it would be nice if you could add a small comment in the `Dockerfile` explaining why we need `BUILDING_DOCKER=yes`.

**Review by jchristgit — COMMENTED** [review]
Solid. Thanks!

**Pull request merged**